src/scheduler/resource_resv.c

```diff
@@ -1490,6 +1491,8 @@ add_resresv_to_array(resource_resv **resresv_arr,
 			return NULL;
 		new_arr[0] = resresv;
 		new_arr[1] = NULL;
+		if (index != NULL)
+		    *index = 0;
 		return new_arr;
 	}
 
```

Comment:
Well, I'm not certain why and @bhroam can explain this better. But, if I have to take a guess, it is because we do not expect a lot of jobs getting added within the cycle to the job arrays (other than in case of creating a subjob). Also, if we overallocated for the future then we will have to keep track of how much have we allocated in the parent object which stores these arrays (sinfo->jobs, qinfo->jobs, sinfo->all_resresv etc.) which makes it more complicated and error-prone.

Reply:
@arungrover is right.  We don't keep track of the length of the array.  When we need the length, we count it.  If the true length of the array was 2x the number of elements in it, we'd still only get 1x the size when we counted.  As Arun said, we shouldn't be calling this function that munch, so it shouldn't be a real big deal.